bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/type/ChannelTypeBuilderTest.java

```diff
@@ -0,0 +1,187 @@
+import org.eclipse.smarthome.core.types.EventOption;
+import org.eclipse.smarthome.core.types.StateDescription;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Tests the {@link ChannelTypeBuilder}.
+ *
+ * @author Stefan Triller - initial contribution
+ *
+ */
+public class ChannelTypeBuilderTest {
+
+    private static final String DESCRIPTION = "description";
+    private static final String ITEM_TYPE = "itemType";
+    private static final String CATEGORY = "category";
+    private static final String LABEL = "label";
+    private static final String TAG = "tag";
+    private static final List<String> TAGS = Arrays.asList("TAG1", "TAG2");
+    private static URI CONFIGDESCRIPTION_URI;
+    private static final String CHANNEL_TYPE_ID = "thingTypeId";
+    private static final String BINDING_ID = "bindingId";
+    private static final StateDescription STATE_DESCRIPTION = new StateDescription(BigDecimal.ZERO, new BigDecimal(100),
+            BigDecimal.ONE, "%s", false, null);
+    private static final EventDescription EVENT_DESCRIPTION = new EventDescription(
+            Arrays.asList(new EventOption(CommonTriggerEvents.DIR1_PRESSED, null),
+                    new EventOption(CommonTriggerEvents.DIR1_RELEASED, null)));
+
+    private ChannelTypeBuilder builder;
+
+    @Before
+    public void setup() throws URISyntaxException {
+        CONFIGDESCRIPTION_URI = new URI("config:dummy");
+        // set up a valid basic ChannelTypeBuilder
+        builder = new ChannelTypeBuilder(BINDING_ID, CHANNEL_TYPE_ID, LABEL);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenChannelTypeIdAndBindingIdBlank_shouldFail() {
+        new ChannelTypeBuilder("", "", LABEL).build();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenChannelTypeIdBlank_shouldFail() {
+        new ChannelTypeBuilder(BINDING_ID, "", LABEL).build();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenBindingIdBlank_shouldFail() {
+        new ChannelTypeBuilder("", CHANNEL_TYPE_ID, LABEL).build();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void whenLabelBlank_shouldFail() {
+        new ChannelTypeBuilder(CHANNEL_TYPE_ID, BINDING_ID, "").build();
+    }
+
+    @Test
+    public void withLabelAndChannelTypeUID_shouldCreateChannelType() {
+        ChannelType channelType = new ChannelTypeBuilder(BINDING_ID, CHANNEL_TYPE_ID, LABEL).withItemType(ITEM_TYPE)
+                .build();
+
+        assertThat(channelType.getUID().getBindingId(), is(BINDING_ID));
+        assertThat(channelType.getUID().getId(), is(CHANNEL_TYPE_ID));
+        assertThat(channelType.getLabel(), is(LABEL));
+    }
+
+    @Test
+    public void withLabelAndThingTypeIdAndBindingID_shouldCreateThingType() {
+        ChannelType channelType = builder.withItemType(ITEM_TYPE).build();
+
+        assertThat(channelType.getUID().getBindingId(), is(BINDING_ID));
```

Comment:
`ITEM_TYPE` is not asserted...

Reply:
`BINDING_ID`, `CHANNEL_TYPE_ID` and `LABEL` are already checked in `withLabelAndChannelTypeUID_shouldCreateChannelType`.